tests/workflow_tests/test_deployment_workflows.py

```diff
@@ -82,3 +85,44 @@ def test_get_workflow_parameters(self):
             }
         }
         self.assertEqual(expected_params, execute_op_workflow.parameters)
+
+    def test_delete_botched_deployment(self):
+
+        from testenv import testenv_instance
+        storage_file_path = os.path.join(
+            testenv_instance.plugins_storage_dir,
+            'agent.json'
+        )
+
+        dsl_path = resource('dsl/botched_deployment.yaml')
+        _id = uuid.uuid1()
+        blueprint_id = 'blueprint_{0}'.format(_id)
+        deployment_id = 'deployment_{0}'.format(_id)
+
+        data = {
+            deployment_id: {'raise_exception_on_delete': True}
+        }
+        with open(storage_file_path, 'w') as f:
+            json.dump(data, f)
+
+        self.client.blueprints.upload(dsl_path, blueprint_id)
+        self.client.deployments.create(blueprint_id, deployment_id)
+
+        execution = self.client.executions.list(
+            deployment_id,
+            True,
+            workflow_id='create_deployment_environment'
+        )[0]
+
+        # cancel deployment creation and wait for it to finish
```

Comment:
the test also cancels the deployment creation to botch it completely
if this is unnecessary, i'll remove it